merlin-framework-processor/src/main/java/gov/nasa/jpl/aerie/merlin/processor/generator/MissionModelGenerator.java

```diff
@@ -336,6 +354,26 @@ public JavaFile generateSchedulerModel(final MissionModelRecord missionModel) {
         .build();
   }
 
+  private CodeBlock.Builder getChildrenStatement(final MissionModelRecord missionModel, final ActivityTypeRecord activityTypeRecord){
+    String[] children = null;
+    //if the effect model is empty, an activity cannot have children
+    if(activityTypeRecord.effectModel().isEmpty()) children = new String[]{};
+    else if(activityTypeRecord.effectModel().get().children().isPresent()){
+      //if children have been declared with an annotation, record those
+      children = activityTypeRecord.effectModel().get().children().get();
+    } else {
+      //if children have not been declared with an annotation, assume the activity might generate all activity types
+      children = missionModel.activityTypes().stream().map(ActivityTypeRecord::name).toArray(String[]::new);
```

Comment:
Memory-wise, rather than returning a list of every activity in the plan, what about wrapping the return in `Optional` and returning `Optional.empty`? It would communicate the same idea-- that `getChildren` is returning the _known_ children and that we don't know what the children are-- without needing a map of 300+ elements with each containing a list of 300+ elements (ie, in the Clipper case).